tests/Unit/PointwiseFunctions/AnalyticData/CurvedWaveEquation/Test_ScalarWaveKerrSchild.cpp

```diff
@@ -0,0 +1,377 @@
+      });
+}
+
+void test_construct_from_options() noexcept {
+  {  // test with wave profile = spherical
+    Options<
+        tmpl::list<ScalarWaveKerr<ScalarWave::Solutions::RegularSphericalWave>>>
+        opts("");
+    opts.parse(
+        "ScalarWaveKerr:\n"
+        "  BlackHoleMass: 0.7\n"
+        "  BlackHoleSpin: [0.1,-0.2,0.3]\n"
+        "  BlackHoleCenter: [0.7,0.6,-2.]\n"
+        "  WaveProfile:\n"
+        "    Gaussian:\n"
+        "      Amplitude: 11.\n"
+        "      Width: 1.4\n"
+        "      Center: 0.3");
+    const auto curved_wave_data_constructed_from_opts =
+        opts.get<ScalarWaveKerr<ScalarWave::Solutions::RegularSphericalWave>>();
+    CHECK(curved_wave_data_constructed_from_opts ==
+          CurvedScalarWave::AnalyticData::ScalarWaveKerrSchild<
+              ScalarWave::Solutions::RegularSphericalWave>(
+              0.7, {{0.1, -0.2, 0.3}}, {{0.7, 0.6, -2.}},
+              std::make_unique<MathFunctions::Gaussian>(11., 1.4, 0.3)));
+
+    // test internals of data constructed from options
+    const gr::Solutions::KerrSchild bh_solution(0.7, {{0.1, -0.2, 0.3}},
+                                                {{0.7, 0.6, -2.}});
+    const ScalarWave::Solutions::RegularSphericalWave flat_wave_solution{
+        std::make_unique<MathFunctions::Gaussian>(11., 1.4, 0.3)};
+
+    test_kerr(curved_wave_data_constructed_from_opts, bh_solution,
+              flat_wave_solution);
+  }
+  {  // test with wave profile = plane
+    Options<tmpl::list<ScalarWaveKerr<ScalarWave::Solutions::PlaneWave<3>>>>
+        opts("");
+    opts.parse(
+        "ScalarWaveKerr:\n"
+        "  BlackHoleMass: 0.7\n"
+        "  BlackHoleSpin: [0.1,-0.2,0.3]\n"
+        "  BlackHoleCenter: [0.7,0.6,-2.]\n"
+        "  WaveVector: [1.0, 1.0, 1.0]\n"
+        "  WaveCenter: [0.0, 0.0, 0.0]\n"
+        "  WaveProfile:\n"
+        "    Gaussian:\n"
+        "      Amplitude: 11.\n"
+        "      Width: 1.4\n"
+        "      Center: 0.3");
+    const auto curved_wave_data_constructed_from_opts =
+        opts.get<ScalarWaveKerr<ScalarWave::Solutions::PlaneWave<3>>>();
+    CHECK(curved_wave_data_constructed_from_opts ==
+          CurvedScalarWave::AnalyticData::ScalarWaveKerrSchild<
+              ScalarWave::Solutions::PlaneWave<3>>(
+              0.7, {{0.1, -0.2, 0.3}}, {{0.7, 0.6, -2.}}, {{1.0, 1.0, 1.0}},
+              {{0.0, 0.0, 0.0}},
+              std::make_unique<MathFunctions::Gaussian>(11., 1.4, 0.3)));
+
+    // test internals of data constructed from options
+    const gr::Solutions::KerrSchild bh_solution(0.7, {{0.1, -0.2, 0.3}},
+                                                {{0.7, 0.6, -2.}});
+    const ScalarWave::Solutions::PlaneWave<3> flat_wave_solution{
+        {{1.0, 1.0, 1.0}},
+        {{0.0, 0.0, 0.0}},
+        std::make_unique<MathFunctions::Gaussian>(11., 1.4, 0.3)};
+
+    test_kerr(curved_wave_data_constructed_from_opts, bh_solution,
```

Comment:
can you do what the Kerr-Schild solution does with `TestHelpers::AnalyticSolutions::test_tag_retrieval` to make sure the interface works correctly?

Reply:
Added tests for tag retrieval